lib/gateways/plan-gateway.js

```diff
@@ -83,4 +82,24 @@ export default class PlanGateway {
       }).map(match => this.get(match))
     );
   }
+
+  async save({ plan }) {
```

Comment:
Is this naming not a bit misleading, as this is not saving the plan - just updating the `goal` value?

Reply:
I imagine we'll end up saving more than just the goal? Not sure as yet

Reply:
Ok, I'm just thinking that if I was maintaining this in future and saw that `PlanGateway` had a `save` method that took a `Plan` instance, I'd expect it to save the entire thing - including any new top level properties I'd added to it.

Typically with DynamoDB I've seen save methods just implemented as complete overwrites, so a `PutItem` with the modified object - unless there was a high likelihood of write contention or need for atomicity when updating a counter etc.

I'm cool with either, but I think if we just do a selective update like we're doing here we should at least add a jsdoc comment to the method stating that it doesn't save the entire object - so people skimming through in the future can get a heads up.

Reply:
The reason I like Update rather than Put here is that it wont save if the plan doesn't exist for whatever reason, that's what create is for. I guess you could argue that having save and create is redundant, I don't think there's ever going to be write contention issues here, updating the existing plan just _feels_ more right to me somehow. Happy with a jsdoc comment. We could also add in the other fields that we might want to update - the title might change for example, the create date wont